packages/lib-classifier/src/store/WorkflowStepStore.js

```diff
@@ -0,0 +1,87 @@
+import { autorun } from 'mobx'
+import { addDisposer, getRoot, types } from 'mobx-state-tree'
+import Step from './Step'
+import { SingleChoiceTask, MultipleChoiceTask } from './Task'
+
+const WorkflowStepStore = types
+  .model('WorkflowStepStore', {
+    active: types.maybe(types.reference(Step)),
+    steps: types.maybe(types.map(Step)),
+    tasks: types.maybe(types.map(types.union((snapshot) => {
+      if (snapshot.type === 'SingleChoiceTask') return SingleChoiceTask;
+      if (snapshot.type === 'MultipleChoiceTask') return MultipleChoiceTask;
+    }, SingleChoiceTask, MultipleChoiceTask)))
+  })
+  .views(
+    get activeStepTasks() {
+      if (self.active) {
+        return self.active.taskKeys.map((taskKey) => {
+          self.tasks.get(taskKey)
+        })
+      }
+
+      return []
+    }
+  )
+  .actions(self => {
+    function afterAttach() {
+      createWorkflowObserver()
+    }
+
+    function reset() {
+      self.active = null
+      self.steps = null
+      self.tasks = null
+    }
+
+    function setStepsAndTasks(workflow) {
+      const taskKeys = Object.keys(workflow.tasks)
+      self.steps = workflow.steps
+      if (taskKeys.length > 0) {
+        taskKeys.forEach((taskKey) => {
+          // Set tasks object as a MobX observable JS map in the store
+          self.tasks.set(taskKey, workflow.tasks[taskKey])
+        })
+      }
+
+      self.selectStep()
+    }
+
+    function createWorkflowObserver() {
+      const workflowDisposer = autorun(() => {
+        const workflow = getRoot(self).workflows.active
+        if (workflow) {
+          self.reset()
+          if (workflow.steps.size > 0) {
```

Comment:
I don't think there would be any steps on the workflow resource here - there would be tasks, but steps would be kept in this store.

Side thought: we could trash the existing workflow store and this one, and have a single store for the workflow which represents it in terms of steps and tasks.

Reply:
I think I'm still working through that conceptually and I just added the existential check just for safety when writing out the skeleton code. It really makes sense to me to have a separate WorkflowStep store, but could be convinced of a single store for the whole workflow resource.

Reply:
Let's leave them as separate stores for now. My feeling was that we don't need a store for the backend representation of the workflow - which doesn't get directly used - but that might change, especially while we're working this out

Reply:
I'm looking over this again and I'm confused about why the workflow resource here wouldn't have any steps. `active` references the `Workflow` model, which should be the workflow resource and I'm proposing the in updated ADR that we add a `steps` property onto the resource itself that is stored in Panoptes. Am I misunderstanding how references in MST work?

Reply:
Ahhhh - I forgot about the proposed resource change, apologies

Reply:
Haha no worries. I reread the code and then the docs and then the code again and confused myself before I realized that I think I'm on the right track here, but wanted to confirm with you. 😆